config/freebsd

```diff
@@ -20,110 +20,121 @@
+
+# check if the compiler_version_info is already set
+if test -z "$cc_version_info"; then
+
+case $CC in
+    # whatever matches *pgcc* will also match *gcc*, so this one must come first
+    *pgcc*)
+        cc_version_info=`$CC $CFLAGS $H4_CFLAGS -V 2>&1 | grep 'pgcc'`
+        ;;
+
+    *gcc*)
+        cc_version_info=`$CC $CFLAGS $H4_CFLAGS --version 2>&1 | grep -v 'PathScale' |\
+            grep 'GCC' | sed 's/\(.*(GCC) [-a-z0-9\. ]*\).*/\1/'`
+        ;;
+
+    *icc*)
+        cc_version_info=`$CC $CCFLAGS $H4_CCFLAGS -V 2>&1 | grep 'Version' |\
+            sed 's/\(Intel.* Compiler\).*\( Version [a-z0-9\.]*\).*\( Build [0-9]*\)/\1\2\3/'`
+        ;;
+
+    *clang*)
+        cc_version_info="`$CC $CFLAGS $H4_CFLAGS --version 2>&1 |\
+            grep 'clang version' | sed 's/.*clang version \([-a-z0-9\.]*\).*/\1/'`"
+        ;;
+
+    *)
+        echo "No match to get cc_version_info for $CC"
+        ;;
 esac
 
-case $F77_BASENAME in 
-  g77)
-    FFLAGS="$FFLAGS"
-    DEBUG_FFLAGS="-g"
-    PROD_FFLAGS="-O"
-    PROFILE_FFLAGS="-pg"
-    ;;
-
-  *)
-    FFLAGS="$FFLAGS"
-    DEBUG_FFLAGS="-g"
-    PROD_FFLAGS="-O"
-    PROFILE_FFLAGS="-pg"
-    ;;
+fi
+
+# get fortran version info
+case $F77 in
+    *gfortran*)
+        fc_version_info=`$F77 $FFLAGS --version 2>&1 |\
+            grep 'GCC' | sed 's/\(.*(GCC) [-a-z0-9\. ]*\).*/\1/'`
+        fc_version="`$F77 $FFLAGS -v 2>&1 |grep 'gcc version' |\
+                sed 's/.*gcc version \([-a-z0-9\.]*\).*/\1/'`"
+        if test X != "X$fc_version"; then
+            fc_version=`echo $fc_version |sed 's/[-a-z]//g'`
+        fi
+        ;;
+
+    *ifc*|*ifort*)
+        fc_version_info=`$F77 $FFLAGS -V 2>&1 | grep 'Version' |\
+            sed 's/\(Intel.* Compiler\).*\( Version [a-z0-9\.]*\).*\( Build [0-9]*\)/\1\2\3/'`
+        fc_version="`$F77 $FFLAGS -V 2>&1 |grep '^Intel'`"
+        if test X != "X$fc_version"; then
+            fc_version="`echo $fc_version |sed 's/.*Version \([-a-z0-9\.\-]*\).*/\1/'`"
+        fi
+        ;;
+
+    *f95*)
+        # Figure out which compiler we are using: pgf90 or Absoft f95
+        RM='rm -f'
+        tmpfile=/tmp/cmpver.$$
+        $F77 -V >$tmpfile
+        if test -s "$tmpfile"; then
+            if( grep -s 'Absoft' $tmpfile > /dev/null) then
+                FC_BASENAME=f95
+            fi
+        fi
+        $RM $tmpfile
+        fc_version_info=`$F77 -V | grep Absoft`
+        ;;
+
+    *g95*|*g77*)
+        fc_version_info=`$F77 $FFLAGS --version 2>&1 |\
+            grep 'GCC'`
+        ;;
+
+    *pgf90*)
+        fc_version_info=`$F77 $FFLAGS -V 2>&1 | grep 'pgf90'`
+        fc_version="`$F77 $FFLAGS -V 2>&1 |grep '^pgf90 '`"
+        if test X != "X$fc_version"; then
+            fc_version=`echo $fc_version |sed 's/pgf90 \([-a-z0-9\.\-]*\).*/\1/'`
+        fi
+        ;;
+
+     *)
+        echo "No match to get fc_version_info for $F77"
+        ;;
 esac
 
+if test X != "X$fc_version"; then
+    # Get the compiler version numbers
```

Comment:
number vs. numbers?

Reply:
I think they say "numbers" because it's often something like "12.1.0" (major.minor.patch)